### PR TITLE
Added Depth First Traversal 

### LINKED ISSUE #1761: [Feature]: Depth First Traversal Algorithm
### Feature Name

Depth First Traversal Algorithm

### Feature Description

This feature involves adding the Depth First Traversal (DFT) algorithm to Algo, allowing users to explore graph traversal in-depth. Depth First Traversal is a fundamental algorithm for searching or traversing through graph data structures. It’s especially useful for applications requiring exhaustive searching, such as pathfinding, cycle detection, and exploring connected components.

### Motivation

Adding Depth First Traversal would greatly benefit users by expanding Algo’s offerings for fundamental algorithms in data structures. It provides users with essential tools for understanding graph-based algorithms, which are widely used in fields like computer science, network analysis, and artificial intelligence.

### Implementation Suggestions (Optional)

Implementation Suggestions
Algorithm Visualization: Implement an interactive DFT visualization that highlights nodes and edges as the algorithm traverses them.
Recursive and Iterative Versions: Offer both recursive and iterative (stack-based) implementations to demonstrate different approaches.
Input Graph Options: Allow users to input their own graphs or select sample graphs to see the algorithm in action.
Documentation and Examples: Provide step-by-step explanations of the traversal process with examples to aid understanding.

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [ ] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [ ] Other

### References (Optional)

_No response_

### PR DESCRIPTION
**Pull request by shivenyadavs**
### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
**Description**
This change introduces the Depth-First Traversal (DFT) algorithm as an addition to the project, providing implementations for preorder, inorder, and postorder traversals in a binary tree structure. This feature enhances the project's tree traversal capabilities, allowing users to explore nodes by going as deep as possible along each branch before backtracking.

**Summary of Changes:**
**Added Depth-First Traversal (DFT) functionality**: Includes recursive implementations for preorder, inorder, and postorder traversal methods.
**C++ Code Implementations:** Provides C++ examples to illustrate each traversal type.

**Motivation and Context**
Depth-First Traversal is a fundamental algorithm used widely in tree-based data structures and search processes. Adding this functionality aligns with project goals of covering essential data structures and algorithms, expanding the use cases for tree traversals in real-world scenarios such as expression evaluation and hierarchical data processing.

Fixes #1761 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shivenyadavs** [comment]
@ajay-dhangar please check the changes

**Comment by shivenyadavs** [comment]
@ajay-dhangar please merge this pr

File: docs/Depth_first_traversal.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    cout << "Inorder Traversal: ";
+    inorderTraversal(root);
+    cout << endl;
+
+    cout << "Postorder Traversal: ";
+    postorderTraversal(root);
+    cout << endl;
+
+    return 0;
+}
```
Comment (ajay-dhangar) [inline docs/Depth_first_traversal.md:100]: close your code example with ```` ``` ````

File: docs/Depth_first_traversal.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+---
+id: dft
+title: Depth-First Traversal (DFT)
```
Comment (ajay-dhangar) [inline docs/Depth_first_traversal.md]: replace `title: Depth-First Traversal (DFT)` with `title: "Depth-First Traversal (DFT)"`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by shivenyadavs** [comment]
please check the changes

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**